lib/main/itoSession.ts

```diff
@@ -0,0 +1,215 @@
+import { ItoMode } from '@/app/generated/ito_pb'
+import { voiceInputService } from './voiceInputService'
+import { recordingStateNotifier } from './recordingStateNotifier'
+import { itoStreamController } from './itoStreamController'
+import { WindowMessenger } from './messaging/WindowMessenger'
+import { TextInserter } from './text/TextInserter'
+import { interactionManager } from './interactions/InteractionManager'
+import { contextGrabber } from './context/ContextGrabber'
+import { grammarRulesService } from './grammar/GrammarRulesService'
+import { getAdvancedSettings } from './store'
+import log from 'electron-log'
+
+export class ItoSession {
```

Comment:
ItoSession is mainly a coordinator class that stays abstract. It orchestrates calls to set up audio streaming, running the grpc stream, pasting the resultant transcript, and handles sending updates to the pill window. It